packages/core/src/client/index.ts

```diff
@@ -79,8 +82,21 @@ export class Web3InboxClient {
     });
   }
 
-  public setAccount(account: string) {
-    Web3InboxClient.clientState.account = account;
+
+  /* Sets account and identity if already registered */
+  public async setAccount(account: string) {
+    // Account setting is duplicated to ensure it only gets updated once
+    // identity state is confirmed
+    if(await this.getAccountIsRegistered(account)) {
+      const identity = await this.notifyClient.identityKeys.getIdentity({ account });
```

Comment:
Could either of those throw ? How would we handle it if an error occurs ?

Reply:
`getAccountIsRegistered` can not throw.

`this.notifyClient.identityKeys.getIdentity` only throws if there is no identity, but `getAccountIsRegistered` prevents that.